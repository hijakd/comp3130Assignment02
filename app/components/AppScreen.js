import React from "react";
import { ImageBackground, Platform, SafeAreaView, StyleSheet } from "react-native";

import AppStyles from "../configs/AppStyles";

const blurRadiusValue = Platform.OS === "android" ? 0.7 : 5.5;

function AppScreen({ children, style }) {
  return (
    <SafeAreaView style={[styles.screen, style]}>
      <ImageBackground
        source={require("../../assets/images/Saguaro-cacti_cropped.jpg")}
        style={styles.screenBackground}
        blurRadius={blurRadiusValue}
      >
        {children} 
      </ImageBackground>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  screen: {
    flex: 1,
  },
  screenBackground: {
    flex: 1,
  },
});

export default AppScreen;
