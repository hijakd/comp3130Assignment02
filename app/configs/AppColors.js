export default {
    black: "#000000",
    white: "#fff",
    primeColor: "#5C755E", // cactus green
    secondaryColor: "#9C9F84",
    sandy: "#F7DCB4",
    brightGreen: "#56D700",
    blazing: "#FF4500",
    blue: "#2323b8",
  };
  
