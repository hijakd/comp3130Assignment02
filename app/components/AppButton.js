import React from 'react';
import { Text, View, StyleSheet, TouchableOpacity } from 'react-native';

import AppColors from '../configs/AppColors';


function AppButton({title, color="primaryColor", textColor = "white", cellPad = 15, onPress}) {
  return (
    <TouchableOpacity onPress={onPress}>
      <View style={[styles.button, {backgroundColor:AppColors[color]},
        {padding: cellPad}]}>
        <Text style={[styles.buttonText, {color: AppColors[textColor]}]}>{title}</Text>
      </View>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    width: "100%",
    justifyContent: "space-around",
    alignItems: "center",
    borderRadius: 20,
    borderStyle: "solid",
    borderWidth: 1,
  },
  buttonText: {
    fontSize: 18,
    textTransform: "uppercase",
    fontWeight: "bold",
  },
});

export default AppButton;
