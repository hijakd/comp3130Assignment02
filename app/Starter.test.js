import React from "react";
import renderer from "react-test-renderer";
// import { jest } from "@jest/globals";

import AppButton from "./components/AppButton";
import AppScreen from "./components/AppScreen";
import AppText from "./components/AppText";
import AppColors from "./configs/AppColors";


/** trivial test to prove testing is running */
test("trivially true", () => {
  expect(1).toBe(1);
});

/** simple test rendering of AppText without props */
test("AppText will render", () => {
  const testVar = renderer.create(<AppText />).toJSON();
  console.log(testVar);
});

/** test rendering of AppText without supplied text */
test("AppText will render", ()=> {
  const testVar = renderer.create(<AppText/>).toJSON();
  // console.log(json.props.style);
  expect(testVar.props.style[0].fontSize).toBe(20);
  expect(testVar.props.style[0].fontFamily).toBe("Avenir-Roman");
});

/** test rendering of AppText containing supplied text */
test("AppText will render", ()=> {
  const testVar = renderer.create(<AppText>Its all Cactus</AppText>).toJSON();
  expect(testVar.props.style[0].fontSize).toBe(20);
  expect(testVar.props.style[0].fontFamily).toBe("Avenir-Roman");
  expect(testVar.children.includes("Its all Cactus"));
});

/**  test rendering of AppButton */
describe('Test custom Button component', ()=> {
  const testVar = renderer.create(<AppButton title="Test"/>);
  it('Should render', ()=> {
    expect(testVar.toJSON()).toBeTruthy();
  });
});

/** test rendering of AppButton with style props supplied */
describe('Test custom Button component', ()=> {
  const testVar = renderer.create(<AppButton title="Test"/>);
  const styles = testVar.toJSON().props.style;
  const {title, color, textColor, cellPadding} = styles;

  it('Should render', ()=> {
    expect(testVar.toJSON()).toBeTruthy();
  });

  it('Should have "primaryColor" backgroundColor set', ()=> {
    expect(color).toBe(AppColors.primaryColor);
  });
});

/** Snapshot of AppScreen component */
it('renders correctly', ()=> {
  const tree = renderer.create(<AppScreen/>).toJSON();
  expect(tree).toMatchSnapshot();
});

