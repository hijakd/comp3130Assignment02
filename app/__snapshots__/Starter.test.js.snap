// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      blurRadius={5.5}
      source={
        Object {
          "testUri": "../../../assets/images/Saguaro-cacti_cropped.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
  </View>
</RCTSafeAreaView>
`;
